### ISSUE TITLE
Inquiry about training retriever model.

### ISSUE THREAD

**Issue by jimmy1016-coder** [richard-peng-xia/MMed-RAG#37]
Hi, thank you for your wonderful work and clear documentation.

I’m a novice in this domain, so I wanted to double-check whether I’m doing things correctly.

I tried to finetune the CLIP model on the Harvard dataset by running the following command as shown in the README:

bash ./scripts/finetune_clip.sh 

![Image](https://github.com/user-attachments/assets/a0cdfc0b-bbca-4182-864c-4672d5b26a9f)

During training, the contrastive loss seems to decrease and converge well. However, when I evaluate on the validation set, the performance is extremely low — the Recall@1 is less than 1%.

Is this expected behavior? Or could there be something wrong with my setup or the way I’m evaluating?

Also, I noticed something strange in the training logs. The iteration jumps from 256/7000 to 6912/7000 almost immediately. I’m using an RTX 4090, and training for 360 epochs takes only about an hour — which seems suspiciously fast.

Is this normal? Or might this indicate an issue with how data loading or epoch iteration is being handled?

Thank you again for your time and help!
![Image](https://github.com/user-attachments/assets/1b195070-5a6e-4fef-acf3-ee838c31025f)

**Comment by richard-peng-xia** [comment]
Hi @jimmy1016-coder,

- For the training speed, it is normal. Don't worry about that.

- As for the evaluation results on the validation set, this depends on the ratio of your data. For example, if your validation set is very large, then the recall will definitely not be significant. However, the quantitative performance on the validation set does not necessarily represent the performance of the retriever. You need to pay attention to the performance of combining MLLM (e.g., LLaVA-Med) with the retrieval information. 

Thanks.

**Comment by qxyqxyqxy642** [comment]
Hello, has your issue been resolved? I’ve encountered the same problem as well—the retriever’s recall rate is extremely low. I’m using the built-in validation set for retrieval.
